### DISCUSSION TITLE
From TRUST4 Output to Data in AIRR format

### DISCUSSION THREAD

**Discussion by lejim** [liulab-dfci/TRUST4#81] [Q&A] [answered]
Hi, 
I am new to BCR Repertoir data analysis. I have used TRUST4, and it worked smoothly on my data. So, thank you for developing this tool! With the output files, I was also able to perform part of the downstream analysis with the Immunarch tool with ease. 
I am struggling to figure out how to convert the TRUST4 output data to AIRR format (https://docs.airr-community.org/en/stable/datarep/rearrangements.html). 
Specifically, I would like to add the information on sequence germline alignment for subsequence mutation and phylogenetic analysis. 
Does anyone have any advice on how to approach this?

**Comment by mourisl** [comment, marked as answer]
Sorry for the delay. I just added the functionality to output in AIRR format in the new release here: https://github.com/liulab-dfci/TRUST4/releases/tag/v1.0.6

For your data, you can run trust4 with the option "--stage 3" to directly generate the AIRR format files from existing results. But I still need some time to work on reporting the germline alignments.

**Reply by snowylxx** [reply]
Hi, your work is so brilliant. I used it for my data research. but here is a question same as the above. You said  we can use existing results to get AIRR format. there are five output files, so which do I use. and how the code be like.
I tried :run-trust4 --stage 3
error: Need to use -b/{-1,-2}/-u to specify input reads.

the bam files are too large so if I rerun, it will take long time.
there is any solution？
thank you a lot!

**Reply by mourisl** [reply]
You still need to put the original options, with an extra option as "--stage 2". (The full AIRR format actually requires some input from stage 2 now in current version.)
